### ISSUE TITLE
LogFile with %date% substitution not creating new file for each day

### ISSUE THREAD

**Issue by Stuart42** [PowershellFrameworkCollective/psframework#531]
Not sure when exactly this started, but here's the behaviour I'm seeing:

-We have several logging providers configured on PowerShell Universal startup, each similar to the code below.
-Log files are created for each provider when the service starts with the date substitution as configured
-Log files written on subsequent days get added to the existing log file. This file, for example contains log events from when it was created until today: "C:\UniversalAzureRepo\Logs\module_logs-2022-06-17.log"

```
$module_logs_paramSetPSFLoggingProvider = @{
    Name                = 'logfile'
    InstanceName        = 'module_logs'
    FilePath            = 'C:\UniversalAzureRepo\Logs\module_logs-%Date%.log'
    IncludeModules      = 'Xibo', 'Nagios', 'AirWatch'
    FileType            = 'cmtrace'
    Enabled             = $true
    LogRotatePath       = 'C:\UniversalAzureRepo\Logs\module_logs-*.log'
    LogRetentionTime    = '365d'
}

Set-PSFLoggingProvider @module_logs_paramSetPSFLoggingProvider
```

Issue seems to be present in 1.7.237.0 and 1.7.235.0

**Comment by FriedrichWeinmann** [comment]
just to be sure I got the correct setting, am I correct in that your issue basically occurs when you configure the logging once and your service keeps running across multiple days without interruption - on days 2+ it will still write to the log of day one?

Also, this is a recent behavior and used to work correctly in versions older than 1.7.235 (which _did_ have some modifications to the system, but those were supposed to not break anything).

**Comment by FriedrichWeinmann** [comment]
Pretty sure it works as you expected in 1.7.227 or older

**Comment by Stuart42** [comment]
I just verified, we were using 1.6.214 before jumping straight to 1.7.235

**Comment by Stuart42** [comment]
That's awesome, thanks Friedrich. PSConfEU sounds like it's a lot of fun. Enjoy!

**Comment by FriedrichWeinmann** [comment]
Sorry about the delay, but finally managed to make the time to nail down the bug.
Released/Resolved in 1.7.244
